apps/ex_nvr/lib/ex_nvr/devices.ex

```diff
@@ -60,6 +60,32 @@ defmodule ExNVR.Devices do
     Device.update_changeset(device, attrs)
   end
 
+  @spec fetch_snapshot(Device.t()) :: binary()
+  def fetch_snapshot(%{stream_config: %{snapshot_uri: nil}}) do
+    {:error, "No value for snapshot uri"}
```

Comment:
It's better to return the reason as an atom and we can format our error response as we like
```elixir
{:error, :no_snapshot_uri}
```